lib/Platform/Intl/java/com/facebook/hermes/intl/PlatformDateTimeFormatterICU.java

```diff
@@ -269,8 +277,21 @@ public void configure(
 
   @RequiresApi(api = Build.VERSION_CODES.N)
   @Override
-  public boolean isValidTimeZone(String timeZone) {
-    return TimeZone.getTimeZone(timeZone).getID().equals(timeZone);
+  public String normalizeValidTimeZone(final String timeZone) throws JSRangeErrorException {
+    Optional<String> normalizedValidTimeZone = allAvailableTimeZones.stream()
+    .filter(new Predicate<String>() {
+      @Override
+      public boolean test(String tz) {
+        return tz.compareToIgnoreCase(timeZone) == 0;
```

Comment:
Use [compareToIgnoreCase](https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#compareToIgnoreCase(java.lang.String)) instead of equalsIgnoreCase to resolve issue described here: https://github.com/facebook/hermes/pull/571#discussion_r723799518